pincer/objects/message/file.py

```diff
@@ -0,0 +1,94 @@
+# Copyright Pincer 2021-Present
+# Full MIT License can be found in `LICENSE` at the project root.
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from io import BytesIO
+import os
+from typing import Optional
+
+from PIL.Image import Image
+
+from ...utils import APIObject
+
+
+@dataclass
+class File(APIObject):
+    """
+    A file that is prepared by the user to be send to the discord
+    API.
+
+    :param content:
+        File bytes.
+
+    :param filename:
+        The name of the file when its uploaded to discord.
+    """
+
+    content: bytes
+    filename: str
+
+    @classmethod
+    def from_file(cls, filepath: str, filename: str = None) -> File:
+        """
+        :param filepath:
+            The path to the file you want to send. Must be string. The file's
+            name in the file path is used as the name when uploaded discord by
+            default.
+
+        :param filename:
+            The name of the file. Will override the default name.
+        """
+
+        with open(filepath, "rb") as data:
+            file = data.read()
+
+        return cls(
+            content=file,
+            filename=filename or os.path.basename(filepath)
+        )
+
+    @classmethod
+    def from_image(
```

Comment:
Could this be renamed to `from_pillow_image`?

Reply:
Sure. I can change this in a few hours.

Reply:
Done